uroot_test.go

```diff
@@ -188,21 +180,34 @@ func testUrootCmdline(t *testing.T) {
 		{
 			name: "include one extra file with rename",
 			args: []string{"-nocmd", "-files=/bin/bash:bin/bush"},
+			env:  []string{"GO111MODULE=off"},
 			validators: []itest.ArchiveValidator{
 				itest.HasFile{"bin/bush"},
 			},
 		},
+	}
+
+	bareTests := []testCase{
 		{
-			name: "make sure dead code gets eliminated",
+			name: "uinitcmd",
+			args: []string{"-uinitcmd=echo foobar fuzz", "-defaultsh=", "github.com/u-root/u-root/cmds/core/init", "github.com/u-root/u-root/cmds/core/echo"},
+			err:  nil,
+			validators: []itest.ArchiveValidator{
+				itest.HasRecord{cpio.Symlink("bin/uinit", "../bbin/echo")},
+				itest.HasContent{
+					Path:    "etc/uinit.flags",
+					Content: "\"foobar\"\n\"fuzz\"",
+				},
+			},
+		},
+		{
+			name: "dead_code_elimination",
 			args: []string{
 				// Build the world + test symbols, unstripped.
 				// Change default shell to gosh for this test as elvish uses the reflect package
-				"-defaultsh=/bbin/gosh", "-build=bb", "-no-strip", "world", "github.com/u-root/u-root/pkg/uroot/test/foo",
+				"-defaultsh=/bbin/gosh", "-no-strip", "world", "github.com/u-root/u-root/pkg/uroot/test/foo",
 				// These are known to disable DCE and need to be exluded.
-				// The reason is https://github.com/golang/go/issues/36021 and is fixed in Go 1.15,
-				// so these can be removed once we no longer support Go < 1.15.
-				"-github.com/u-root/u-root/cmds/exp/builtin",
-				"-github.com/u-root/u-root/cmds/exp/run",
+				// elvish uses reflect.Value.Call and is expected to not use DCE.
```

Comment:
nit: maybe not accurate to claim so. It reads to me that only certain code elimination are inhibited, https://github.com/u-root/u-root/issues/1477. I think DCE covers more than un-used public methods.

Reply:
maybe it is true as commented -- I am just speculating